File: cs50x/psets/6/credit/credit.py
from cs50 import get_string


def main():
    cc = get_string("Number: ")

    if isLegit(cc):
        check_type(cc)
    else:
        print("INVALID")


def isLegit(num):
    sum = 0
    for i, m in enumerate(reversed(num)):
        m = int(m)
        if i % 2 == 0:
            sum += m
        else:
            for j in str(m * 2):
                sum += int(j)

    if sum % 10 == 0:
        return True
    else:
        return False


def check_type(cc):
    # get first 2 digit with slicing
    first2 = int(cc[0:2])
    length = len(cc)

    if (first2 == 34 or first2 == 37) and length == 15:
        print("AMEX")
    elif 50 < first2 <= 55 and length == 16:
        print("MASTERCARD")
    elif 40 <= first2 < 50 and (length == 13 or length == 16):
        print("VISA")
    else:
        print("INVALID")


if __name__ == "__main__":
    main()

File: cs50x/psets/1/credit/credit.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    long cardNum;
    do
    {
        cardNum = get_long("Number: ");
    }
    while (cardNum < 0);

    int length = 0;
    int temp, arr[16];
    while (cardNum > 0)
    {
        temp = cardNum % 10;
        arr[length] = temp;

        cardNum /= 10;
        length++;
    }

    int sum, isLegit;
    int multiply = 0;
    int notMultiply = 0;

    for (int i = 0; i < length; i++)
    {
        if (i % 2 == 0)
        {
            notMultiply += arr[i];
        }
        else
        {
            int temp1;
            int temp2;

            temp1 = arr[i] * 2;
            while (temp1 > 0)
            {
                temp2 = temp1 % 10;
                multiply += temp2;
                temp1 /= 10;
            }
        }
    }

    sum = multiply + notMultiply;
    isLegit = sum % 10;

    int one = length - 1;
    int two = length - 2;
    if (isLegit == 0 && length == 15 && arr[one] == 3 && (arr[two] == 4 || arr[two] == 7))
    {
        printf("AMEX\n");
    }
    else if (isLegit == 0 && length == 16 && arr[one] == 5 && (arr[two] <= 5 && arr[two] > 0))
    {
        printf("MASTERCARD\n");
    }
    else if (isLegit == 0 && (length == 13 || length == 16) && arr[one] == 4)
    {
        printf("VISA\n");
    }
    else
    {
        printf("INVALID\n");
    }
}